dpppt-backend-sdk/dpppt-backend-sdk-ws/src/main/java/org/dpppt/backend/sdk/ws/security/NoValidateRequest.java

```diff
@@ -16,9 +18,13 @@ public boolean isValid(Object authObject) {
 	}
 
 	@Override
-	public long getKeyDate(Object authObject, Object others) {
+	public long getKeyDate(Object authObject, Object others) throws InvalidDateException {
 		if (others instanceof ExposeeRequest) {
-			return ((ExposeeRequest) others).getKeyDate();
+			ExposeeRequest request = ((ExposeeRequest) others);
+			if(request.getKeyDate() < OffsetDateTime.now().minusDays(21).toInstant().toEpochMilli()) {
```

Comment:
if (request.getKeyDate() > System.currentTimeMillis()) {
				if (maxKeyDate > System.currentTimeMillis()) {						throw new InvalidDateException();
					// the maximum key date is the current day.					} 

dürfte man hier wohl auch noch einfügen.

Reply:
ich nehme an du meinst request.getKeyDate()? maxKeyDate gibts nicht mehr

Reply:
Habs mal angepasst und noch ein paar Tests hinzugefügt